tests/unit/core/components/TestAuthenticateButton.js

```diff
@@ -132,4 +111,44 @@ describe('<AuthenticateButton />', () => {
     }));
     expect(mapStateToProps(store.getState()).isAuthenticated).toEqual(true);
   });
+
+  it('allows a signed-in user to log out', () => {
+    const { store } = dispatchSignInActions();
+    const handleLogOut = sinon.stub();
+    const allProps = {
+      handleLogOut,
+      i18n: getFakeI18nInst(),
+    };
+
+    const wrapper = shallowUntilTarget(
+      <AuthenticateButton store={store} {...allProps} />,
+      AuthenticateButtonBase,
+    );
+
+    wrapper.simulate('click', createFakeEvent());
+
+    sinon.assert.calledWith(handleLogOut, {
+      api: store.getState().api,
+    });
+  });
+
+  describe('createHandleLogOutFunction()', () => {
+    it('logs out a signed-in user on the server and client sides', () => {
+      sinon.stub(api, 'logOutFromServer').returns(Promise.resolve());
+
+      const { store, state } = dispatchSignInActions();
+      const dispatchSpy = sinon.spy(store, 'dispatch');
+
+      const handleLogOut = createHandleLogOutFunction(dispatchSpy);
```

Comment:
Heh, I had to looks this up in the docs. I was going to suggest passing `store.dispatch` since the intention is a bit more clear but they are identical so it's fine either way.